### PR TITLE
[Draft & TEST] SQL server connector

### PR DESCRIPTION
**Pull request by shenqianjin**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/ServerDependComboBox.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package com.microsoft.azure.toolkit.intellij.connector.database.component;
+
+import com.microsoft.azure.toolkit.intellij.common.AzureComboBox;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+public class ServerDependComboBox<S, T> extends AzureComboBox<T> {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/ServerDependComboBox.java:15]: weird class

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/SpringDatasourceLineMarkerProvider.java [hunk 2]
```diff
@@ -50,15 +56,16 @@
...
+            .map(c -> ((DatabaseResourceConnection) c))
             .filter(c -> StringUtils.equals(envPrefix, c.getResource().getEnvPrefix()))
             .findAny()
-            .map(MySQLDatabaseResourceConnection::getResource)
+            .map(DatabaseResourceConnection::getResource)
             .map(r -> new LineMarkerInfo<>(
                 element, element.getTextRange(),
                 AzureIconLoader.loadIcon(AzureIconSymbol.MySQL.BIND_INTO),
-                element2 -> String.format("Connect to Azure Database for MySQL (%s)", r.getJdbcUrl().getServerHost()),
+                element2 -> String.format("Connect to %s (%s)", DatabaseResource.Definition.getTitleByType(r.getType()), r.getJdbcUrl().getServerHost()),
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/SpringDatasourceLineMarkerProvider.java:68]: use message bundle instead.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/Connection.java [hunk 3]
```diff
@@ -52,4 +55,10 @@ default boolean isApplicableFor(@NotNull RunConfiguration configuration) {
      */
     default void updateJavaParametersAtRun(RunConfiguration configuration, @NotNull JavaParameters parameters) {
     }
+
+    default String getType() {
+        final String resourceType = Objects.nonNull(getResource()) && StringUtils.isNotBlank(getResource().getType()) ? getResource().getType() : "default";
+        final String consumerType = Objects.nonNull(getConsumer()) && StringUtils.isNotBlank(getConsumer().getType()) ? getConsumer().getType() : "default";
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/Connection.java:61]: try `Optional`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/ServerDependComboBox.java (hunk 1 above)
Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/connector/database/component/ServerDependComboBox.java:15]: please comment other's PRs (they are split all code changes of this PR as different PRs). this PR is only drafted for testing.

**Pull request closed without merging**